tutorial/creating_packages/build_packages.rst

```diff
@@ -1,2 +1,220 @@
-Build packages
-==============
\ No newline at end of file
+Build packages: the build() method
+==================================
+
+.. important::
+
+    In this example, we retrieve the *fmt* and *gtest* Conan packages from a Conan
+    repository with packages compatible with Conan 2.0. To run this example successfully
+    you should add this remote to your Conan configuration (if did not already do it)
+    doing: ``conan remote add conanv2
+    https://conanv2beta.jfrog.io/artifactory/api/conan/conan --index 0``
+
+
+We already used a Conan recipe that has a ``build()`` method and learned how to use that
+to invoke a build system and build our packages. In this tutorial, we will modify that
+method and explain how you can use it to do things like building and running tests,
+conditional patching the source code or select the build system you want to use
+conditionally.
+
+Please, first clone the sources to recreate this project. You can find them in the
+`examples2.0 repository <https://github.com/conan-io/examples2>`_ on GitHub:
+
+.. code-block:: bash
+
+    $ git clone https://github.com/conan-io/examples2.git
+    $ cd examples2/tutorial/creating_packages/build_method
+
+
+Build and run tests for your project
+------------------------------------
+
+You will notice some changes in the **conanfile.py** file from the previous recipe.
+Let's check the relevant parts:
+
+.. code-block:: python
+    :emphasize-lines: 12, 19, 27-28, 35-36
+
+    class helloRecipe(ConanFile):
+        name = "hello"
+        version = "1.0"
+
+        ...
+
+        def source(self):
+            git = Git(self)
+            git.clone(url="https://github.com/conan-io/libhello.git", target=".")
+            # Please, be aware that using the head of the branch instead of an inmutable tag
+            # or commit is not a good practice in general
+            git.checkout("with_tests")
+
+        ...
+
+        def requirements(self):
+            if self.options.with_fmt:
+                self.requires("fmt/8.1.1")
+            self.test_requires("gtest/1.11.0")
+
+        ...
+
+        def generate(self):
+            tc = CMakeToolchain(self)
+            if self.options.with_fmt:
+                tc.variables["WITH_FMT"] = True
+            if not self.conf.get("tools.build:skip_test", default=False):
```

Comment:
We made this built-in in CMake integration, for the cmake standard BUILD_TESTING variable. It is good to use a custom one to explain, but maybe add a comment to let readers know.